### ISSUE TITLE
[Feature Request]  Show messages from previous session without internet connection 

### ISSUE THREAD

**Issue by KaKooDa** [telegramdesktop/tdesktop#26018]
### Is your feature request related to a problem?

Unlike Telegram on mobile, Telegram doesn't load and won't show you anything unless you have a working internet connection. That is specially troublesome if Telegram is censored in your country and you need to pick of something from telegram!

### Describe the solution you'd like

I'd like Telegram desktop to behave like Mobile one! Showing previous messages even if you are offline (the messages that were sent to you last time you where online). Then, if you had internet connection, update to show new messages!

### Describe alternatives you've considered

Saving every and each message and file before closing Telegram, which is really troublesome.

**Comment by Aokromes** [comment]
https://github.com/telegramdesktop/tdesktop/issues/1087

**Comment by KaKooDa** [comment]
@Aokromes 
These issues aren't the same! I'm asking for something slightly different. Please read what I've written carefully

**Comment by john-preston** [comment]
@KaKooDa This may become possible if the app will make a transition to TDLib. This may be kept as a feature request, I guess.

**Comment by Aokromes** [comment]
> @Aokromes These issues aren't the same! I'm asking for something slightly different. Please read what I've written carefully

it's the same, hability to go offline being allowed to read existing messages.

**Comment by john-preston** [comment]
Not really, they want a way to disable network in app while it's running, while here a local messages database is requested. #1087 can be implemented without a local messages database and local messages database can be implemented without an explicit offline mode being offered in the app.

**Comment by KaKooDa** [comment]
@Aokromes 
Excuse me, I think I couldn't explain what I mean clearly. By "offline" I mean "Not having internet connection" and online means "Having a working internet connection".
Here is an example:
Assume that you have opened telegram yesterday and didn't do so until today. Now, if you open Telegram on a mobile device (i.e Android) you can see your old messages (sent yesterday or any time before) even if you are "offline", However you can't see newer messages (messages from yesterday until today) unless you become "online". On the other hand, if you use Telegram desktop, you can't see any messages at all (Even the old messages) unless you become "online".

Thanks

**Comment by Aokromes** [comment]
> @Aokromes Excuse me, I think I couldn't explain what I mean clearly. By "offline" I mean "Not having internet connection" and online means "Having a working internet connection". Here is an example: Assume that you have opened telegram yesterday and didn't do so until today. Now, if you open Telegram on a mobile device (i.e Android) you can see your old messages (sent yesterday or any time before) even if you are "offline", However you can't see newer messages (messages from yesterday until today) unless you become "online". On the other hand, if you use Telegram desktop, you can't see any messages at all (Even the old messages) unless you become "online".
> 
> Thanks

then it's the exact same request:

"I would like to request an ability to disconnect client without quitting completely, like all other good messengers have.
By right-clicking on taskbar icon, user should be able to switch between online and offline status.
While offline, client shouldn't be able to send/receive any network requests and generally behave like there's no Internet connection at all."

**Comment by Shjpr9** [comment]
> > @Aokromes Excuse me, I think I couldn't explain what I mean clearly. By "offline" I mean "Not having internet connection" and online means "Having a working internet connection". Here is an example: Assume that you have opened telegram yesterday and didn't do so until today. Now, if you open Telegram on a mobile device (i.e Android) you can see your old messages (sent yesterday or any time before) even if you are "offline", However you can't see newer messages (messages from yesterday until today) unless you become "online". On the other hand, if you use Telegram desktop, you can't see any messages at all (Even the old messages) unless you become "online".
> > Thanks
> 
> then it's the exact same request:
> 
> "I would like to request an ability to disconnect client without quitting completely, like all other good messengers have. By right-clicking on taskbar icon, user should be able to switch between online and offline status. While offline, client shouldn't be able to send/receive any network requests and generally behave like there's no Internet connection at all."

That's not similar at all! As he says:

> Not really, they want a way to disable network in app while it's running, while here a local messages database is requested. #1087 can be implemented without a local messages database and local messages database can be implemented without an explicit offline mode being offered in the app.

We are looking for a local database to store the messages received from previous sessions and show them in Telegram even if you are offline and don't have access to internet.

Currently, you don't have access to previous messages without going online. But in Telegram Mobile you can see all your messages and groups without having access to the internet

![image](https://user-images.githubusercontent.com/67644313/225276379-ce479947-c502-4bae-b802-a5db390f5bee.png)

**Comment by VadimVadimVadim1337** [comment]
True

**Comment by NTMan** [comment]
This can be especially annoying if Telegram is blocked and proxies for bypassing blocking are stored in one of the Telegram chats. Chicken and egg problem.

![Screenshot from 2024-02-27 15-21-54](https://github.com/telegramdesktop/tdesktop/assets/200750/47412f42-5518-4394-bc0f-4c39534d6d74)
